exporter/metrics_test.go

```diff
@@ -16,3 +19,54 @@ func TestSanitizeMetricName(t *testing.T) {
 		}
 	}
 }
+
+func TestRegisterConstHistogram(t *testing.T) {
+	exp := getTestExporter()
+
+	metricName := "foo"
+
+	ch := make(chan prometheus.Metric)
+	go func() {
+		exp.registerConstHistogram(ch, metricName, []string{"bar"}, 12, .24, map[float64]uint64{}, "test")
+		close(ch)
+	}()
+
+	for m := range ch {
+		if strings.Contains(m.Desc().String(), metricName) {
+			return
+		}
+	}
+	t.Errorf("Histogram was not registered")
+}
+
+func TestFindOrCreateMetricsDescriptionFindExisting(t *testing.T) {
+	exp := getTestExporter()
+	exp.metricDescriptions = make(map[string]*prometheus.Desc)
+
+	metricName := "foo"
+	metricDesc := "bar"
+	labels := []string{"1", "2"}
+
+	description := newMetricDescr("ns", metricName, metricDesc, labels)
+	exp.metricDescriptions[metricName] = description
```

Comment:
Another way to test this without manipulating the `metricDescriptions` directly would be to call `findOrCreateMetricDescription()` twice and check that you still only have one entry in metricDescriptions (len(metricDescriptions) == 1).